contracts/contracts/Tally.sol

```diff
@@ -154,59 +116,60 @@ contract Tally is Ownable, SnarkCommon, CommonUtilities, Hasher, DomainObjs {
     uint256 tallyBatchSize = TREE_ARITY ** intStateTreeDepth;
     uint256 batchStartIndex = tallyBatchNum * tallyBatchSize;
 
-    // save some gas because we won't overflow uint256
-    unchecked {
-      tallyBatchNum++;
-    }
-
     (uint256 numSignUps, ) = poll.numSignUpsAndMessages();
 
     // Require that there are untallied ballots left
     if (batchStartIndex >= numSignUps) {
       revert AllBallotsTallied();
     }
 
-    bool isValid = verifyTallyProof(_proof, numSignUps, batchStartIndex, tallyBatchSize, _newTallyCommitment);
+    uint256[] memory circuitPublicInputs = getPublicCircuitInputs(_newTallyCommitment);
```

Comment:
you already have numSignups, batchStartIndex, tallyBatchSize in this scope, can you pass the directly to verifyTallyProof?

Reply:
So, here is the same problem and idea as for MP.